### PR TITLE
feat: use framework.ExpectNotEqual in e2e test

### LINKED ISSUE #79861: Use framework.ExpectNotEqual() for e2e tests
After fixing golint failures under test/e2e, we need to put gomega for each Expect() and Equal().
Then common error checking code has become like:

```
gomega.Expect(foo).NotTo(gomega.Equal(bar))
gomega.Expect(foo).ToNot(gomega.Equal(bar))
```

We can replace this code with framework.ExpectNotEqual(foo, bar) which is more readable and easy to be understood what the code does.

/assign
/cc @oomichi 
/sig testing

### PR DESCRIPTION
**Pull request by draveness**
/kind cleanup
/sig testing
/assign @oomichi 

**What this PR does / why we need it**:

use framework.ExpectNotEqual in e2e test

**Which issue(s) this PR fixes**:

Fixes #79861

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/apps/statefulset.go [hunk 1]
```diff
@@ -1130,9 +1127,10 @@ func rollbackTest(c clientset.Interface, ns string, ss *appsv1.StatefulSet) {
 	framework.ExpectNoError(err)
 	ss = e2esset.WaitForStatus(c, ss)
 	currentRevision, updateRevision = ss.Status.CurrentRevision, ss.Status.UpdateRevision
-	gomega.Expect(currentRevision).NotTo(gomega.Equal(updateRevision),
-		"Current revision should not equal update revision during roll back")
 	framework.ExpectEqual(priorRevision, updateRevision, "Prior revision should equal update revision during roll back")
+	framework.ExpectNotEqual(currentRevision, updateRevision, "Current revision should not equal update revision during roll back")
+	gomega.Expect(priorRevision).To(gomega.Equal(updateRevision),
```
Comment (andrewsykim) [inline test/e2e/apps/statefulset.go]: why this change?

Comment (andrewsykim) [reply test/e2e/apps/statefulset.go]: looks like this is same as line 1130 right?

Comment (oomichi) [reply test/e2e/apps/statefulset.go]: Nice catch, line 1132 should be removed I think also.

File: test/e2e/apimachinery/table_conversion.go [hunk 2]
```diff
@@ -110,9 +110,9 @@ var _ = SIGDescribe("Servers with support for Table transformation", func() {
 			Do().Into(pagedTable)
 		framework.ExpectNoError(err, "failed to get pod templates in Table form in namespace: %s", ns)
 		gomega.Expect(len(pagedTable.Rows)).To(gomega.Equal(2))
```
Comment (oomichi) [inline test/e2e/apimachinery/table_conversion.go]: Can we merge https://github.com/kubernetes/kubernetes/pull/79685 first before this?

File: test/e2e/framework/util.go [hunk 3]
```diff
@@ -1352,6 +1352,11 @@ func ExpectEqual(actual interface{}, extra interface{}, explain ...interface{})
 	gomega.Expect(actual).To(gomega.Equal(extra), explain...)
 }
 
+// ExpectNotEqual expects the specified two are not the same, otherwise an exception raises
```
Comment (oomichi) [inline test/e2e/framework/util.go]: Nice function :-)

File: test/e2e/instrumentation/logging/generic_soak.go [hunk 4]
```diff
@@ -24,7 +24,6 @@ import (
 	"time"
 
 	"github.com/onsi/ginkgo"
-	"github.com/onsi/gomega"
```
Comment (oomichi) [inline test/e2e/instrumentation/logging/generic_soak.go]: Cool, thanks for doing that.

**Review by oomichi — COMMENTED** [review]
Thanks for doing this!
Just few comments.

**Comment by oomichi** [comment]
This seems to be rebased.

File: test/e2e/apimachinery/table_conversion.go (hunk 2 above)
Comment (draveness) [reply test/e2e/apimachinery/table_conversion.go]: Sure :)

**Comment by draveness** [comment]
All the comments have been addresses, please take another look

**Comment by draveness** [comment]
/retest

Regards,
Draven

File: test/e2e/framework/util.go [hunk 5]
```diff
@@ -1352,6 +1352,11 @@ func ExpectEqual(actual interface{}, extra interface{}, explain ...interface{})
...
+func ExpectNotEqual(actual interface{}, extra interface{}, explain ...interface{}) {
```
Comment (andrewsykim) [inline test/e2e/framework/util.go]: I know we are just being consistent with `ExpectEqual` here, but curious why it's called `extra` instead of something like `expected`?
